cli/args/flags.rs

```diff
@@ -2395,6 +2396,12 @@ fn publish_subcommand() -> Command {
           .help("Allow publishing with slow types")
           .action(ArgAction::SetTrue),
       )
+      .arg(
+        Arg::new("provenance")
+          .long("provenance")
+          .help("From CI/CD system, publicly links the package to where it was built and published from.")
+          .action(ArgAction::SetTrue)
+      )
```

Comment:
We can also do this in a follow up, but can we enable provenance by default when we are using GH OIDC for auth?

And then add a `--no-provenance` option to force it off, and `--provenance` option to force it on.

We can then also use `--provenance` to do enable provenance for local publishes at some later date.